src/runtime_src/core/common/utils.cpp

```diff
@@ -186,6 +194,22 @@ unit_convert(size_t size)
   return ret+" "+unit[i-1];
 }
 
+std::string
+format_base10_shiftdown3(uint64_t value)
+{
+  static double DECIMAL_SHIFT = 1000.0;
+  static int DIGIT_PRECISION = 3;
+  return precision(static_cast<double>(value) / DECIMAL_SHIFT, DIGIT_PRECISION);
+}
+
+std::string
+format_base10_shiftdown6(uint64_t value)
+{
+  static double DECIMAL_SHIFT = 1000000.0;
+  static int DIGIT_PRECISION = 6;
```

Comment:
```
constexpr double decimal_shift = 1000000.0;
constexpr int digit_precision = 6;
```
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es45-avoid-magic-constants-use-symbolic-constants
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#con5-use-constexpr-for-values-that-can-be-computed-at-compile-time

Reply:
done nd thanks for references